src/pineko/cli/convolute.py

```diff
@@ -51,13 +50,26 @@ def subcommand(
     PDF is an optional PDF set compatible with the EKO to compare grid and FK table.
     """
     grid = pineappl.grid.Grid.read(grid_path)
-    operators = eko.io.legacy.load_tar(op_path)
+    operators = eko.EKO.edit(op_path)
     # This solution is temporary: the theory card will be available in the eko
     tcard_path = pathlib.Path(tcard_path)
     with open(tcard_path, encoding="utf-8") as f:
         theory_card = yaml.safe_load(f)
-    new_tcard = eko.compatibility.update_theory(theory_card)
-    astrong = sc.Couplings.from_dict(new_tcard)
+    legacy_class = eko.io.runcards.Legacy(theory_card, {})
+    new_tcard = legacy_class.new_theory
+    evmod = eko.io.types.CouplingEvolutionMethod.EXACT
+    if theory_card["ModEv"] == "TRN":
+        evmod = eko.io.types.CouplingEvolutionMethod.EXPANDED
+    quark_masses = [(x.value) ** 2 for x in new_tcard.quark_masses]
+    sc = eko.couplings.Couplings(
```

Comment:
you should move this stuff down to evolve else it is repeated here and in theory - because now we can move it down, because now the EKO object contains everything - same holds for the preparation of operators (see also Alessandro's comment above)